go/parquet/pqarrow/file_writer.go

```diff
@@ -134,6 +134,15 @@ func (fw *FileWriter) RowGroupTotalBytesWritten() int64 {
 	return 0
 }
 
+// RowGroupNumRows returns the number of rows written to the current row group.
+// Returns an error if they are unequal between columns that have been written so far.
+func (fw *FileWriter) RowGroupNumRows() (int, error) {
+	if fw.rgw != nil {
+		return fw.rgw.NumRows()
+	}
+	return 0, nil
+}
+
```

Comment:
If we're going to add a method for the number of rows written to the current row group, should we also add a pass through to calling `FileWriter.NumRows` to allow getting the total number of rows written to the file in general?

Reply:
Sounds good, @zeroshade.  Adding tests for a `NumRows` method revealed that the `file.Writer` doesn't increment the number of rows written when appending a new row group writer.

I've added a commit that updates the `file.Writer` so it tracks the number of rows written across multiple row groups and adds a `NumRows` method on `pqarrow.FileWriter` that accesses this.

Let me know if it would be preferable to ticket the `file.Writer` num rows issue separately and create a dedicated fix for that.  The assertion added here fails on the default branch: https://github.com/apache/arrow/pull/38507/commits/c30d6dd4ae443f86f5835fae62f163ffa373cdf1#diff-f4e1b3867fa5f621a41484668fbce7194ecdb289f68fb6ed0e112e8e3a00bde8R100

Reply:
I put up #38517 with just the fix for the `file.Writer` issue.

Reply:
@zeroshade - I've rebased this after #38517, so think it is ready for review.